app/assets/javascripts/discourse/app/lib/sidebar/common/community-section/everything-section-link.js

```diff
@@ -51,7 +51,7 @@ export default class EverythingSectionLink extends BaseSectionLink {
   }
 
   get currentWhen() {
-    return "discovery.latest discovery.new discovery.unread discovery.top";
+    return "discovery.latest discovery.new discovery.unread discovery.top discovery.hot discovery.unseen discovery.posted";
```

Comment:
Maybe we can add the `discovery` service and use `this.discovery.onDiscoveryRoute` instead? It would double highlight for categories and tags, but it's not entirely wrong as they're all "Topics" – but to avoid the confusion we can `return this.discovery.onDiscoveryRoute && !this.discovery.category && !this.discovery.tag`